### PR TITLE
add pluggable io

### PR DESCRIPTION
**Pull request by pohmelie**

### CODE REVIEW

File: nil/storage/hash/interface.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    @abstractmethod
+    async def put_payload(self, payload: str) -> str:
+        """
+        Put payload to hash storage and return transaction hash
+        """
+
+    async def get_payload(self, tx: str) -> str:
+        """
+        Get payload by transaction hash
+        """
```
Comment (nickgashkov) [inline nil/storage/hash/interface.py]: Shouldn't this method be abstract as well?

File: nil/node.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+
+@click.command(context_settings=dict(auto_envvar_prefix="NIL"))
+@click.option("--log-level", show_default=True, default="INFO")
+@click.option("--web-host", show_default=True, default="localhost")
+@click.option("--web-port", show_default=True, default=8080, type=int)
+@click.option("--io-network-max-workers", show_default=True, default=100, type=int)
+@click.option("--io-file-max-workers", show_default=True, default=5, type=int)
+# data storage
+@click.option("--data-storage-type", show_default=True, default="sqlite")
```
Comment (nickgashkov) [inline nil/node.py]: Maybe add option validation, given, that this value is used in `importlib.import_module` later on?

```suggestion
@click.option("--data-storage-type", show_default=True, default="sqlite", type=click.Choice(["sqlite"]))
```

File: nil/node.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+@click.option("--data-storage-sqlite-dsn", show_default=True, default="sqlite:///data.sqlite")
+# hash storage
+@click.option("--hash-storage-type", show_default=True, default="sqlite")
```
Comment (nickgashkov) [inline nil/node.py]:
```suggestion
@click.option("--hash-storage-type", show_default=True, default="sqlite", type=click.Choice(["sqlite"]))
```

File: nil/node.py [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+@click.option("--hash-storage-sqlite-dsn", show_default=True, default="sqlite:///hash.sqlite")
+# network
+@click.option("--network-type", show_default=True, default="http")
```
Comment (nickgashkov) [inline nil/node.py]:
```suggestion
@click.option("--network-type", show_default=True, default="http", type=click.Choice(["http"]))
```

File: nil/api.py [hunk 5]
```diff
@@ -5,7 +5,9 @@
 from loguru import logger
 from yarl import URL
 
-from .interface import AbstractApi, AbstractDataStorage, AbstractHashStorage, Message, UserMessage
+from .interface import AbstractApi, Message, UserMessage
+from .storage.data.interface import AbstractDataStorage
+from .storage.hash.interface import AbstractHashStorage
 
 
 @dataclass
```
Comment (nickgashkov) [inline nil/api.py]: Correct me, if I'm wrong, but it looks like `nil.api` has been replaced with `nil.network` and can be safely removed.

File: nil/storage/hash/interface.py (hunk 1 above)
Comment (pohmelie) [reply nil/storage/hash/interface.py]: For sure

File: nil/api.py (hunk 5 above)
Comment (pohmelie) [reply nil/api.py]: Yes, my bad. Lost in refactoring forest :)

**Pull request merged**